src/services/VirtualFileSystemService.spec.ts

```diff
@@ -1,532 +1,423 @@
     });
   });
 
   describe('#writeFile', () => {
     it('correctly writes to the file', async () => {
+      const mockFilePath = path.resolve('/notExistingMockFile.ts');
+
       const stringToWrite = "const toWrite = 'written';";
 
-      await service.writeFile('/non-existing-file.ts', stringToWrite);
+      await virtualFileSystemService.writeFile(mockFilePath, stringToWrite);
 
-      const finalContent = await service.readFile('/non-existing-file.ts');
+      const finalContent = await virtualFileSystemService.readFile(mockFilePath);
       expect(finalContent).toEqual(stringToWrite);
 
-      await service.deleteFile('/non-existing-file.ts');
+      await virtualFileSystemService.deleteFile(mockFilePath);
     });
 
     it('correctly rewrites a file', async () => {
-      const stringToWrite = "const toRewrite = 'rewritten';";
-      const previousContent = await service.readFile('/src/index.ts');
-
-      await service.writeFile('/src/index.ts', stringToWrite);
-
-      const finalContent = await service.readFile('/src/index.ts');
-      expect(finalContent).toEqual(stringToWrite);
-
-      await service.writeFile('/src/index.ts', previousContent);
-    });
+      const mockFileToRewrite = path.resolve('/mockFolder/mockFileToRewrite.ts');
 
-    it('correctly writes to a symbolic link', async () => {
-      const stringToWrite = "const toWrite = 'written';";
-      const previousContent = await service.readFile('/src/index.ts');
-
-      await service.writeFile('/src/indexSL.ts', stringToWrite);
-
-      const finalContent = await service.readFile('/src/index.ts');
-      expect(finalContent).toEqual(stringToWrite);
-
-      await service.writeFile('/src/index.ts', previousContent);
-    });
-
-    it('correctly writes to a broken symbolic link', async () => {
-      const stringToWrite = "const toWrite = 'written';";
+      const stringToWrite = "const toRewrite = 'rewritten';";
+      const previousContent = await virtualFileSystemService.readFile(mockFileToRewrite);
 
-      await service.writeFile('/src/indexSLbroken.ts', stringToWrite);
+      await virtualFileSystemService.writeFile(mockFileToRewrite, stringToWrite);
 
-      const finalContent = await service.readFile('/src/non-existing-file.ts');
+      const finalContent = await virtualFileSystemService.readFile(mockFileToRewrite);
       expect(finalContent).toEqual(stringToWrite);
 
-      await service.deleteFile('/src/non-existing-file.ts');
+      await virtualFileSystemService.writeFile(mockFileToRewrite, previousContent);
     });
 
     it("throws an error if the parent directory doesn't exist", async () => {
-      await expect(service.writeFile('/non/existing/file.ts', '...')).rejects.toThrow('is not a directory');
-    });
+      const mockFilePath = path.resolve('/notExistingMockFolder/file.ts');
 
-    it('throws an error if the parent directory is not a directory', async () => {
```

Comment:
Why?

Reply:
It doesn't matter for VFSS it the path is folder or directory. It can create folder called "index.ts". I will try it once more, but I think this is the reason.